### PR TITLE
fix: sensitive data in events

### PR DESCRIPTION
**Pull request by exu**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by rangoo94 — COMMENTED** [review]
* Emitter don't need `Envs` passed then to `NewEmitter`
* Will webhooks have access to `Envs` then?

**Comment by vsukhin** [comment]
Might be, we need to leave onyl WEBHOOKS_ prefixed env

**Comment by rangoo94** [comment]
> Might be, we need to leave onyl WEBHOOKS_ prefixed env

Will it work, if the User will have some secrets exposed in API Server and required for specific webhooks (like Slack's API token)? Or maybe we should just populate `Envs` in the place where webhook is actually sent, and avoid it there?

**Comment by vsukhin** [comment]
That would be better

**Comment by exu** [comment]
> * Emitter don't need `Envs` passed then to `NewEmitter`
> * Will webhooks have access to `Envs` then?

Do we want to expose all envs to some external webhook calls at all?

**Comment by rangoo94** [comment]
> Do we want to expose all envs to some external webhook calls at all?

Probably better would be to have them whitelisted by some prefix (as @vsukhin mentioned), but at the moment Users are already using it with no patterns, so it would be breaking change to add prefix 😕

File: pkg/event/kind/webhook/listener_test.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func TestWebhookListener_Notify(t *testing.T) {
 		svr := httptest.NewServer(testHandler)
 		defer svr.Close()
 
-		l := NewWebhookListener("l1", svr.URL, "", testEventTypes, "", "", nil, false, nil, nil, nil, v1.NewMetrics(), nil)
+		l := NewWebhookListener("l1", svr.URL, "", testEventTypes, "", "", nil, false, nil, nil, nil, v1.NewMetrics(), nil, map[string]string{})
```
Comment (rangoo94) [inline pkg/event/kind/webhook/listener_test.go:37]: `nil` would be fine too (here and everywhere below)

**Review by vsukhin — APPROVED** [review]

File: pkg/event/kind/webhook/listener_test.go (hunk 1 above)
Comment (exu) [reply pkg/event/kind/webhook/listener_test.go]: done

**Comment by exu** [comment]
> LGTM, but it would be great to test it 👍

Will check it on sandbox

**Comment by exu** [comment]
image

webhooks are there on sandbox

**Pull request merged**